### PR TITLE
libhdhomerun: git -> 20200521

### PR DESCRIPTION
**Pull request by LouisDK1**
###### Motivation for this change

New release 20200521 which is not available on Github: https://github.com/Silicondust/libhdhomerun

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by LouisDK1** [comment]
I'm a bit unsure why this fails.

When I run `nix-shell -p nixpkgs-review --run "nixpkgs-review wip` I get this error:

`error: patch failed: pkgs/development/libraries/libhdhomerun/default.nix:1
error: pkgs/development/libraries/libhdhomerun/default.nix: patch does not apply
Failed to apply diff in /home/louis/.cache/nixpkgs-review/rev-c2e6ac97a41835d77084b7623b49ec181884c26b-dirty-10/nixpkgs`

Commenting the patchPhase and 3 preceding lines does not help.

**Comment by LouisDK1** [comment]
Running "nix-shell -p nixpkgs-review --show-trace --run "nixpkgs-review pr 93671" works. Installing package locally on my NixOS works.

@titanous Can you check if this package will build on Darwin? It seems like it wants to build with i386 flags which shouldn't be supported on 10.15+.

**Comment by titanous** [comment]
I don't have a Darwin machine easily available at the moment, someone else will have to test.

File: pkgs/development/libraries/libhdhomerun/default.nix [hunk 1]
```diff
@@ -1,12 +1,12 @@
-{ stdenv, fetchgit }:
+{ stdenv, fetchurl }:
 
-stdenv.mkDerivation {
-  name = "libhdhomerun-1efbcb";
+stdenv.mkDerivation rec {
+  pname = "libhdhomerun";
+  version = "20200521";
```
Comment (sikmir) [inline pkgs/development/libraries/libhdhomerun/default.nix:5]:
```suggestion
  version = "2020-05-21";
```

File: pkgs/development/libraries/libhdhomerun/default.nix [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 
-  src = fetchgit {
-    url = "git://github.com/Silicondust/libhdhomerun.git";
-    rev = "1efbcb2b87b17a82f2b3d873d1c9cc1c6a3a9b77";
-    sha256 = "11iyrfs98xb50n9iqnwfphmmnn5w3mq2l9cjjpf8qp29cvs33cgy";
+  src = fetchurl {
+    url = "https://download.silicondust.com/hdhomerun/libhdhomerun_${version}.tgz";
```
Comment (sikmir) [inline pkgs/development/libraries/libhdhomerun/default.nix:8]:
```suggestion
    url = "https://download.silicondust.com/hdhomerun/libhdhomerun_${stdenv.lib.replaceStrings [ "-" ] [ ""] version}.tgz";
```

File: pkgs/development/libraries/libhdhomerun/default.nix (hunk 1 above)
Comment (LouisDK1) [reply pkgs/development/libraries/libhdhomerun/default.nix:5]: If you look at https://www.silicondust.com/support/downloads/ and the ubuntu and arch repos they're all referring to the release as YYYYMMDD. To comply with upstream I think we should keep it that way.

File: pkgs/development/libraries/libhdhomerun/default.nix (hunk 2 above)
Comment (LouisDK1) [reply pkgs/development/libraries/libhdhomerun/default.nix:8]: As stated before I prefer to stick with the official version.

**Comment by LouisDK1** [comment]
@rycee This package build for me on NixOS but I'm unsure about Darwin (macOS). Should I make it a Linux-only package or what would you suggest?

**Comment by rycee** [comment]
I'd suggest asking in #nixos @ freenode if there is any Darwin user that could help out by building the package. Basically it should be sufficient to run
```console
$ nix run nixpkgs.nix-review --command nix-review pr 93671
```
in a Nixpkgs checkout

**Comment by sikmir** [comment]
Darwin user is here)
```sh
$ nix run nixpkgs.nix-review --command nix-review pr 93671
...
1 package marked as broken and skipped:
kodiPlugins.pvr-hdhomerun

1 package built:
libhdhomerun
...
```

**Comment by LouisDK1** [comment]
> Darwin user is here.
> Result of `nixpkgs-review pr 93671` [1](https://github.com/Mic92/nixpkgs-review)
> 1 package marked as broken and skipped:
> - kodiPlugins.pvr-hdhomerun
> 1 package built:
> - libhdhomerun

Thanks. Does kodiPlugins.pvr-hdhomerun have to be fixed for this package to get merged?

**Comment by rycee** [comment]
Since Kodi seems to be limited to Linux I can't imagine it being critical that it's plugins build on Darwin. Rebased to master in 526c85966461ddca08897db85a181b4b9d7c1e11.

**Pull request closed without merging**